src/StatusQ/Controls/StatusTagSelector.qml

```diff
@@ -0,0 +1,121 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.12
+import QtQuick.Controls 2.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+//TODO check colors, sizes, margins, general clean up
+Item {
+    id: root
+
+    implicitWidth: 448
+    implicitHeight: 44
+
+    property alias textEdit: edit
+    property alias text: edit.text
+    property string warningText: ""
+    property int nameCountLimit: 5
+    property ListModel namesModel: ListModel { }
+
+    function find(model, criteria) {
+      for (var i = 0; i < model.count; ++i) if (criteria(model.get(i))) return model.get(i);
+      return null;
+    }
+
+    function insertTag(name, id) {
+        if (!find(namesModel, function(item) { return item.publicId === id }) && namesModel.count < root.nameCountLimit) {
+            namesModel.insert(namesModel.count, {"name": name, "publicId": id});
+            edit.clear();
+        }
+    }
+
+    Rectangle {
+        anchors.fill: parent
+        radius: 8
+        color: Theme.palette.baseColor2
+        border.width: 1
+        border.color: edit.activeFocus ? Theme.palette.primaryColor2 : "transparent"
+
+        RowLayout {
+            anchors.fill: parent
+            anchors.leftMargin: 16
+            anchors.rightMargin: 16
+            spacing: 8
+            StatusBaseText {
+                Layout.preferredWidth: 22
+                Layout.alignment: Qt.AlignVCenter | Qt.AlignLeft
+                color: Theme.palette.baseColor1
+                text: qsTr("To: ")
+            }
+
+            ListView {
+                id: namesList
+                //TODO check if 220 is the correct limit for width
+                Layout.preferredWidth: (childrenRect.width > 220) ? 220 : childrenRect.width
+                implicitHeight: 30
+                visible: (count > 0)
+                Layout.alignment: Qt.AlignVCenter
+                model: namesModel
+                orientation: ListView.Horizontal
+                spacing: 8
+                clip: true
+                onContentWidthChanged: {
+                    positionViewAtEnd();
+                }
+
+                delegate: Rectangle {
```

Comment:
Realising now that this is the actual tag component.
I wonder if we should make some sort of button out of that. At, least, it should have proper hover effects/visuals (even if only on the "X" icon).

Reply:
this is the delegate, correct. There is no hovered state defined on Figma for this nevertheless, the `StatusIconTabButton` is used for the close icon which already has a hovered effect.